src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -512,6 +515,36 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Allows the database type to be customized if we
+        /// are not able to properly infer the database type.
+        /// </summary>
+        /// <param name="dbDataParameter">The parameter.</param>
+        /// <param name="parameterInfo"></param>
+        internal void CustomizeParameterType(IDbDataParameter dbDataParameter, DatabaseParameterInfo parameterInfo)
```

Comment:
Proposal: rename to "SetParamValue" and add `p.Value = stringValue` (line 504) in this method